Sources/HTTP/HTTPVersion.swift

```diff
@@ -29,12 +29,44 @@ extension HTTPVersion : Hashable {
     public static func == (lhs: HTTPVersion, rhs: HTTPVersion) -> Bool {
         return lhs.hashValue == rhs.hashValue
     }
-
```

Comment:
Hash values by design are guaranteed to be equal for two equal values but not vice versa.
So it is better to replace implementation of `==` with `lhs.major == rhs.major && lhs.minor == rhs.minor`.

For example, in current implementation hash values for versions 1.1 and 2.2 are equal. I suggest to implement `hashValue` in a more collision-resistant way, something like: `(major << 8) | minor`